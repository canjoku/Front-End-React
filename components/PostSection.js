import { Component } from 'react'

import Markdown from 'markdown-to-jsx'

class PostSection extends Component {
  render () {
    const { body, createdAt, title} = this.props.post
    return (
      <div className='c-post-section'>
        <h1 className='c-post-section__primary-heading u-bottom-margin-medium '>{title}</h1>
        <p className='c-post-section__body u-bottom-margin-medium'><Markdown>{body}</Markdown></p>
      </div>
    )
  }
}

export default PostSection