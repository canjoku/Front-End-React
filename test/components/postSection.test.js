import PostSection from '../../components/PostSection'

test('<PostSection /> renders correctly', () => {
  const post =
  {
    title: "title of second post",
    body: "body of second post",
    createdAt: "18-01-2019"
  }
  const component = shallow(<PostSection post={post}/>)
  expect(component).toMatchSnapshot()
})