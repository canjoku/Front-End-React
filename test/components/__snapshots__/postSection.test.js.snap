// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostSection /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostSection
    post={
      Object {
        "body": "body of second post",
        "createdAt": "18-01-2019",
        "title": "title of second post",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="c-post-section__primary-heading u-bottom-margin-medium "
        >
          title of second post
        </h1>,
        <p
          className="c-post-section__body u-bottom-margin-medium"
        >
          <Markdown>
            body of second post
          </Markdown>
        </p>,
      ],
      "className": "c-post-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "title of second post",
          "className": "c-post-section__primary-heading u-bottom-margin-medium ",
        },
        "ref": null,
        "rendered": "title of second post",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Markdown>
            body of second post
          </Markdown>,
          "className": "c-post-section__body u-bottom-margin-medium",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "body of second post",
          },
          "ref": null,
          "rendered": "body of second post",
          "type": [Function],
        },
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="c-post-section__primary-heading u-bottom-margin-medium "
          >
            title of second post
          </h1>,
          <p
            className="c-post-section__body u-bottom-margin-medium"
          >
            <Markdown>
              body of second post
            </Markdown>
          </p>,
        ],
        "className": "c-post-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "title of second post",
            "className": "c-post-section__primary-heading u-bottom-margin-medium ",
          },
          "ref": null,
          "rendered": "title of second post",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Markdown>
              body of second post
            </Markdown>,
            "className": "c-post-section__body u-bottom-margin-medium",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "body of second post",
            },
            "ref": null,
            "rendered": "body of second post",
            "type": [Function],
          },
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
